File: server/src/main/java/de/ingrid/igeserver/api/RolesApiController.java
package de.ingrid.igeserver.api;

import de.ingrid.igeserver.model.Role;
import de.ingrid.igeserver.model.Role1;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "/api")
public class RolesApiController implements RolesApi {


    public ResponseEntity<Void> createRole(String id, Role1 role) {
        // do some magic!
        return new ResponseEntity<Void>(HttpStatus.OK);
    }

    public ResponseEntity<Void> deleteRole(String id) {
        // do some magic!
        return new ResponseEntity<Void>(HttpStatus.OK);
    }

    public ResponseEntity<String> getRole(String id) {
        return ResponseEntity.ok("{ \"id\": -1 }");
    }

    public ResponseEntity<String> listRoles() {
        return ResponseEntity.ok("[]");
    }

    public ResponseEntity<Void> updateRole(String id, Role role) {
        // do some magic!
        return new ResponseEntity<Void>(HttpStatus.OK);
    }

}

File: server/src/main/java/de/ingrid/igeserver/api/RolesApiController.kt
package de.ingrid.igeserver.api

import de.ingrid.igeserver.model.Role
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = ["/api"])
class RolesApiController : RolesApi {
    override fun createRole(id: String, role: Role): ResponseEntity<Void> {
        // do some magic!
        return ResponseEntity(HttpStatus.OK)
    }

    override fun deleteRole(id: String): ResponseEntity<Void> {
        // do some magic!
        return ResponseEntity(HttpStatus.OK)
    }

    override fun getRole(id: String): ResponseEntity<String> {
        return ResponseEntity.ok("{ \"id\": -1 }")
    }

    override fun listRoles(): ResponseEntity<String> {
        return ResponseEntity.ok("[]")
    }

    override fun updateRole(id: String, role: Role): ResponseEntity<Void> {
        // do some magic!
        return ResponseEntity(HttpStatus.OK)
    }
}